aten/src/ATen/native/ChanelShuffle.cpp

```diff
@@ -11,6 +12,13 @@ namespace at {
 namespace native {
 
 Tensor channel_shuffle(const Tensor& self, int64_t groups) {
+#if defined(USE_XNNPACK)
+  if (self.is_contiguous(MemoryFormat::ChannelsLast) &&
+      xnnpack::use_channel_shuffle(self, groups)) {
+    return xnnpack::channel_shuffle(self, groups);
+  }
+#endif
+
```

Comment:
Let's move all this after line 33 where all the checks are done. The checks should be common for both.